src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,109 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import React from 'react';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const checkForEmptyField = () => {
+    if (!title || !imgUrl || !imdbUrl || !imdbId) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const resetAll = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+
+    setCount(last => last + 1);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newMovie: Movie = {
+      title: title,
+      description: description,
+      imgUrl: imgUrl,
+      imdbUrl: imdbUrl,
+      imdbId: imdbId,
+    };
+
+    if (checkForEmptyField()) {
+      return;
+    }
+
+    onAdd(newMovie);
+    resetAll();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={newTitle => setTitle(newTitle)}
```

Comment:
code of TextField.tsx allows us to use just a `setState` function

TextField.tsx:
`onChange={event => onChange(event.target.value)}`

```suggestion
        onChange={setTitle}
```